srtcore/window.h

```diff
@@ -229,26 +230,64 @@ class CPktTimeWindow: CPktTimeWindowTools
        m_LastArrTime = m_CurrArrTime;
    }
 
-   /// Record the arrival time of the first probing packet.
+   /// Shortcut to test a packet for possible probe 1 or 2
+   void probeArrival(const CPacket& pkt)
+   {
+       const int inorder16 = pkt.m_iSeqNo & PUMASK_SEQNO_PROBE;
+
+       // for probe1, we want 16th packet
+       if (inorder16 == 0)
+       {
+           probe1Arrival(pkt);
+       }
+
+       // for probe2, we want 17th packet
+       if (inorder16 == 1)
+       {
+           probe2Arrival(pkt);
+       }
+   }
 
-   void probe1Arrival()
+   /// Record the arrival time of the first probing packet.
+   void probe1Arrival(const CPacket& pkt)
    {
        m_ProbeTime = CTimer::getTime();
+       m_Probe1Sequence = pkt.m_iSeqNo; // Record the sequence where 16th packet probe was taken
    }
 
    /// Record the arrival time of the second probing packet and the interval between packet pairs.
 
-   void probe2Arrival(int pktsz = 0)
+   void probe2Arrival(const CPacket& pkt)
    {
+       // Reject probes that don't refer to the very next packet
+       // towards the one that was lately notified by probe1Arrival.
+       // Otherwise the result can be stupid.
+
+       // Simply, in case when this wasn't called exactly for the
+       // expected packet pair, behave as if the 17th packet was lost.
+
+       if (    // probe1Arrival wasn't called yet, no start point then
+               m_Probe1Sequence == -1
+               // not very next towards lately recorded 16th
+               || CSeqNo::incseq(m_Probe1Sequence) != pkt.m_iSeqNo
+          )
```

Comment:
What happened with formatting?

Reply:
I wanted to somehow add comments by `//` to the particular condition being tested.

Reply:
Please move the comments out of the if() statement. It is not very hard to reword the comment so that it can correlate to relevant parts of the conditions within the if().